### PR TITLE
fix `The replaceState function of the history is overridden incorrectly` (close #1146)

### LINKED ISSUE #1146: The replaceState function of the history is overridden incorrectly
Example for reprodusing:
```javascript
function HistoryUrl () {
    this.absoluteURL = "http://localhost:5000/";
    this.requestURL = "http://localhost:5000/";
}

HistoryUrl.prototype.toString = function () {
    return this.absoluteURL;
};

history.replaceState({ some: 8334 }, null, new HistoryUrl());
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

File: src/client/sandbox/node/window.js [hunk 1]
```diff
@@ -268,19 +268,19 @@ export default class WindowSandbox extends SandboxBase {
         };
 
         if (typeof window.history.pushState === 'function' && typeof window.history.replaceState === 'function') {
-            window.history.pushState = function () {
-                if (typeof arguments[2] === 'string')
-                    arguments[2] = getProxyUrl(arguments[2]);
+            var createHistoryStateManipulationWrapper = function (nativeFn) {
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: `createWrapperForHistoryStateManipulationFn`

File: test/client/fixtures/sandbox/node/window-test.js [hunk 2]
```diff
@@ -76,18 +76,82 @@ test('parameters passed to the native function in its original form', function (
...
-
     var documentFragment = document.createDocumentFragment();
 
     checkNativeFunctionArgs('querySelector', 'documentFragmentQuerySelector', documentFragment);
     checkNativeFunctionArgs('querySelectorAll', 'documentFragmentQuerySelectorAll', documentFragment);
     checkNativeFunctionArgs('dispatchEvent', 'windowDispatchEvent', window);
 });
 
+if (window.history.replaceState) {
+    asyncTest('replace history state with a various types of url', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/window-test.js]: You changed `replaceState` and `pushState` functions.
So, need to add test cases for `pushState` function

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: test/client/fixtures/sandbox/node/window-test.js [hunk 3]
```diff
@@ -76,18 +76,86 @@ test('parameters passed to the native function in its original form', function (
...
+                var iframeHistory       = iframeWindow.history;
+                var iframeNativeMethods = iframeHammerhead.nativeMethods;
+                var iframeLocation      = iframeWindow.location;
+                var baseUrl             = 'http://' + location.host + '/some/path';
+
+                iframeHammerhead.get('./utils/url-resolver').updateBase(baseUrl, iframe.contentDocument);
+                iframeHammerhead.get('./utils/destination-location')
+                    .forceLocation('http://' + iframeLocation.host + '/sessionId/' + baseUrl);
+
+                var testUrl = function (url, fnName, nativeFn) {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/window-test.js]: Les't change the second function parameter to `fn` and pass overriden function respectively.

File: test/client/fixtures/sandbox/node/window-test.js [hunk 4]
```diff
@@ -76,18 +76,86 @@ test('parameters passed to the native function in its original form', function (
...
-
     var documentFragment = document.createDocumentFragment();
 
     checkNativeFunctionArgs('querySelector', 'documentFragmentQuerySelector', documentFragment);
     checkNativeFunctionArgs('querySelectorAll', 'documentFragmentQuerySelectorAll', documentFragment);
     checkNativeFunctionArgs('dispatchEvent', 'windowDispatchEvent', window);
 });
 
+if (window.history.replaceState) {
+    asyncTest('replace history state with a various types of url', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/window-test.js]: Rename test name to the `window.history.replaceState, window.history.pushState`

**Comment by LavrovArtem** [comment]
FPR

**Pull request merged**